src/DropzoneJsUploadSave.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\dropzonejs\DropzoneJsUploadSave.
+ */
+
+namespace Drupal\dropzonejs;
+
+use Drupal\Core\Entity\EntityManagerInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\Core\Render\RendererInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\file\FileInterface;
+use Symfony\Component\HttpFoundation\File\MimeType\MimeTypeGuesserInterface;
+use Symfony\Component\Validator\Constraints\File;
+use Drupal\Core\File\FileSystemInterface;
+
+/**
+ * A service that saves files uploaded by the dropzonejs element as file
+ * entities.
+ *
+ * Most of this file mimics or directly copies what core does. For more
+ * information and comments see file_save_upload().
+ */
+class DropzoneJsUploadSave implements DropzoneJsUploadSaveInterface {
+
+  /**
+   * Entity manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityManagerInterface
+   */
+  protected $entityManager;
+
+  /**
+   * Mime type guesser service.
+   *
+   * @var \Symfony\Component\HttpFoundation\File\MimeType\MimeTypeGuesserInterface;
+   */
+  protected $mimeTypeGuesser;
+
+  /**
+   * The file system service.
+   *
+   * @var \Drupal\Core\File\FileSystemInterface;
+   */
+  protected $fileSystem;
+
+  /**
+   * The logger service.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactoryInterface
+   */
+  protected $logger;
+
+  /**
+   * The renderer service.
+   *
+   * @var \Drupal\Core\Render\RendererInterface
+   */
+  protected $renderer;
+
+  /**
+   * Construct the DropzoneUploadSave object.
+   *
+   * @param \Drupal\Core\Entity\EntityManagerInterface $entity_manager
+   *   Entity manager service.
+   * @param \Symfony\Component\HttpFoundation\File\MimeType\MimeTypeGuesserInterface $mimetype_guesser
+   *   The mime type guesser service.
+   * @param \Drupal\Core\File\FileSystemInterface $file_system
+   *   The file system service.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $logger_factory
+   *   The logger factory service.
+   * @param Drupal\Core\Render\RendererInterface $renderer
+   *   The renderer service.
+   */
+  public function __construct(EntityManagerInterface $entity_manager, MimeTypeGuesserInterface $mimetype_guesser, FileSystemInterface $file_system, LoggerChannelFactoryInterface $logger_factory, RendererInterface $renderer) {
+    $this->entityManager = $entity_manager;
+    $this->mimeTypeGuesser = $mimetype_guesser;
+    $this->fileSystem = $file_system;
+    $this->logger = $logger_factory->get('dropzonejs');
+    $this->renderer = $renderer;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function saveFile($uri, $destination, $extensions, AccountProxyInterface $user) {
```

Comment:
Only thing that needs to be public is this IMO. Everything else can be protected (and not part of interface as such).